### DISCUSSION TITLE
Exposing a CMake Subproject to a Meson Dependency

### DISCUSSION THREAD

**Discussion by ChristianJacobsen** [mesonbuild/meson#11823] [Q&A]
Hi!

Can I provide a CMake Subproject with a dependency from Meson?

The case here is the library [opentelemetry-cpp](https://github.com/open-telemetry/opentelemetry-cpp), which uses the CMake build system.
When compiled using the option `WITH_OTLP` it depends on Protobuf, gRPC, and the JSON library by Niels Lohmann.
Since my wrap file for `opentelemetry-cpp` uses the tar-ball as a source (I know I can alter this, but bear with me), I "don't have access to" the Git submodules the project uses for its dependencies on the libraries mentioned above. I would also like to provision them using Meson for the dependency fetching/resolution.

The question then becomes: how do I let CMake know where to find these libraries?

For Protobuf I can specify `Protobuf_LIBRARIES` and `Protobuf_INCLUDE_DIR` using the options passable to the `cmake.subproject()` function, but getting the location from the dependency object (returned from my call to `dependency('protobuf')`) seems impossible, as no method on the dependency object returns the library path or include directory path. I believe the other libraries' dependencies are solvable in the same fashion.

I know I'm probably not going about this most straightforwardly, and maybe the better approach would've been for me to write a `meson.build` file for `opentelemetry-cpp` instead. The only problem is that I don't particularly want to maintain said file, nor do I think I will get it 100% correct.

Is what I'm asking about here impossible, or am I simply unable to find the correct page in the documentation?

Thanks in advance for your help ❤️

**Comment by eli-schwartz** [comment]
This sounds like essentially #8104

**Reply by ChristianJacobsen** [reply]
That's unfortunate, as the PR hasn't seen much activity in the past couple years...
